libjps/include/jps.h

```diff
@@ -0,0 +1,101 @@
+#ifndef _JPS_H_
+#define _JPS_H_
+
+#include <stdbool.h>
+#include <stdlib.h>
+
+typedef struct {
+    void * sim_context;
+} SimulationContext;
+
+typedef struct {
+    bool has_error;
+    char * error_message;
+} Status;
+
+typedef struct {
+    double x;
+    double y;
+} Coordinate;
+
+typedef struct {
+    int id;
+    Coordinate coordinate;
+    double velocity;
+    double v_phi;
+} Agent;
+
+typedef struct {
+    Agent * agents;
+    size_t size;
+} Agents;
+
+typedef struct {
+    int * goals;
+    size_t num_golas;
+} Route;
+
+/**
+ * @name JPSlib Functions.
+ * @param [in]  simulation_context  The simulation context (owned by JPSlib).
+ * @param [out] status              Status object containing error information and message.
+ * @{
+ **/
+
+/**
+ * Initialize the simulation
+ * @param [in]  geometry_file The filepath to the geometry configuration file.
+ * @return                    The simulation context.
+ **/
+extern SimulationContext initialize(const char * geometry_file, Status * status);
+
+/**
+ * Adds a new agent to the pedestrian simulation.
+ *
+ * @param [in] agent_id           The unique identifier for the newly created agent.
+ * @param [in] starting_area      The unique identifier of the starting area to place the new agent.
+ * @param [in] route              The goals of the pedestrian reflecting the desired route.
+ * @return                        Bool whether the agent could be added to the designated area.
+ **/
+extern bool transfer_agent(
+    SimulationContext simulation_context,
+    int agent_id,
+    int starting_area,
+    Route route,
+    Status * status);
+
+/**
+ * Computes the trajectories for all pedestrians for a given simulation_time.
+ *
+ * @param [in]  simulation_time Simulation time duration to compute.
+ **/
+extern void simulate(SimulationContext simulation_context, double simulation_time, Status * status);
+
+/**
+ * Receives the agents from the simulation containing position and velocity.
+ *
+ * @return Agents array containing coordinates and velocities.
+ **/
+extern Agents get_agents(SimulationContext simulation_context, Status * status);
+
+/**
+ * Retrieves the agents which are removed from the pedestrian simulation.
+ *
+ * @returns An array of agents.
+ **/
+extern Agents get_retrievable_agents(SimulationContext simulation_context, Status * status);
```

Comment:
Whats the logic behind 'retrive'? Since I did not partake in any discussion about the architecture please excuse the stupid question: Why can't the simulation simply remove agents that are in an exit zone? Nd ofc notify the caller of the agents that 'left' and where they left?